diff --git a/actions/modify.action.inc b/actions/modify.action.inc
index 71b6181..19d8e55 100644
--- a/actions/modify.action.inc
+++ b/actions/modify.action.inc
@@ -146,7 +146,20 @@ function views_bulk_operations_modify_action_form($context, &$form_state) {
         '#title' => $property['label'],
       );
 
-      $determined_type = ($property['type'] == 'boolean') ? 'checkbox' : 'textfield';
+      switch ($property['type']) {
+        case 'boolean':
+          $determined_type = 'checkbox';
+          break;
+        case 'list':
+          $determined_type = 'checkboxes';
+          break;
+        case 'integer':
+          $determined_type = 'select';
+          break;
+        default:
+          $determined_type = 'textfield';
+      }
+
       $form['properties'][$key] = array(
         '#type' => $determined_type,
         '#title' => $property['label'],
