diff --git a/og.module b/og.module
index dac0ab7..94bffdb 100755
--- a/og.module
+++ b/og.module
@@ -935,7 +935,7 @@ function _og_update_entity_fields($entity_type, $entity) {
     else {
       $target_type = $field['settings']['target_type'];
       $gids = og_get_entity_groups($entity_type, $entity, array(), $field_name);
-      $gids = !empty($gids[$target_type]) ? $gids[$target_type] : array();
+      $gids = !empty($gids[$target_type]) ? array_values($gids[$target_type]) : array();
     }
     if ($gids) {
       $wrapper->{$field_name}->set($gids);
