diff --git a/rules.install b/rules.install
index 8485df3..6259d81 100644
--- a/rules.install
+++ b/rules.install
@@ -513,16 +513,18 @@ function rules_update_7213() {
  * Switch out the rules_event_whitelist variable for a cache equivalent.
  */
 function rules_update_7214() {
-  // Set new event_whitelist cache cid.
-  rules_set_cache('rules_event_whitelist', variable_get('rules_event_whitelist', array()));
-  // Delete old conf variable.
-  variable_del('rules_event_whitelist');
-  // Avoid any missing class errors.
-  registry_rebuild();
-  // Clear and rebuild Rules caches.
-  // See: rules_admin_settings_cache_rebuild_submit.
-  rules_clear_cache();
-  rules_get_cache();
-  _rules_rebuild_component_cache();
-  RulesEventSet::rebuildEventCache();
+  if (module_exists('rules')) {
+    // Set new event_whitelist cache cid.
+    rules_set_cache('rules_event_whitelist', variable_get('rules_event_whitelist', array()));
+    // Delete old conf variable.
+    variable_del('rules_event_whitelist');
+    // Avoid any missing class errors.
+    registry_rebuild();
+    // Clear and rebuild Rules caches.
+    // See: rules_admin_settings_cache_rebuild_submit.
+    rules_clear_cache();
+    rules_get_cache();
+    _rules_rebuild_component_cache();
+    RulesEventSet::rebuildEventCache();
+  }
 }
