diff --git a/modules/comment/views_handler_field_comment_link.inc b/modules/comment/views_handler_field_comment_link.inc
index 162924e..d63a6ce 100644
--- a/modules/comment/views_handler_field_comment_link.inc
+++ b/modules/comment/views_handler_field_comment_link.inc
@@ -48,9 +48,13 @@ class views_handler_field_comment_link extends views_handler_field_entity {
 
   function render_link($data, $values) {
     $text = !empty($this->options['text']) ? $this->options['text'] : t('view');
-    $comment = $this->get_value($values);
-    $nid = $comment->nid;
-    $cid = $comment->cid;
+    if ($comment = $this->get_value($values)) {
+      $nid = $comment->nid;
+      $cid = $comment->cid;
+    }
+    else {
+      $nid = $cid = NULL;
+    }
 
     $this->options['alter']['make_link'] = TRUE;
     $this->options['alter']['html'] = TRUE;
diff --git a/modules/comment/views_handler_field_comment_link_edit.inc b/modules/comment/views_handler_field_comment_link_edit.inc
index 0b06c0e..3458591 100644
--- a/modules/comment/views_handler_field_comment_link_edit.inc
+++ b/modules/comment/views_handler_field_comment_link_edit.inc
@@ -34,7 +34,7 @@ class views_handler_field_comment_link_edit extends views_handler_field_comment_
     parent::render_link($data, $values);
     // ensure user has access to edit this comment.
     $comment = $this->get_value($values);
-    if (!comment_access('edit', $comment)) {
+    if (empty($comment) || !comment_access('edit', $comment)) {
       return;
     }
 
